saq/queue/postgres.py

```diff
@@ -613,10 +620,11 @@ async def _dequeue(self) -> list[Job]:
                 },
             )
             results = await cursor.fetchall()
-        for result in results:
-            job = self.deserialize(result[0])
-            if job:
-                jobs.append(job)
+            for result in results:
+                job = self.deserialize(result[0])
+                if job:
+                    await self.update(job, status=Status.ACTIVE, connection=conn)
```

Comment:
Update the job's status to match the table column status in the same transaction to keep them in sync.